ecs-cli/modules/utils/compose/ecs_params_reader.go

```diff
@@ -98,6 +119,87 @@ func (cd *ContainerDef) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+// HealthCheck.UnmarshalYAML is a custom unmarshaler for healthcheck that parses
+// both the docker compose and ecs inspired syntaxes
+func (h *HealthCheck) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	// All of the different health check formats are mutually exclusive in each of their fields
+	// This makes parsing simple.
+	healthCheck := HealthCheck{}
```

Comment:
I'm a little confused why we are instantiating this here and setting fields on it versus dealing with the `healthCheckFormat` data structure -- seems like there are two things happening in this function, unmarshalling and setting.

Reply:
The custom unmarshaller is supposed to unmarshal to a format struct, and then set it on the main struct- if you look at the customer unmarshaller in ECS Params for ContainerDef, it does the same thing- both unmarshalling into one struct, and then setting the main struct.